src/android/com/outsystems/plugins/healthfitness/OSHealthFitness.kt

```diff
@@ -218,21 +220,26 @@ class OSHealthFitness : CordovaImplementation() {
     }
 
     private fun setBackgroundJob(args: JSONArray) {
-        // save arguments for later use in case we need to request permissions
+        // save arguments for later use
         backgroundParameters = gson.fromJson(args.getString(0), BackgroundJobParameters::class.java)
-        if (!notificationPermissions.hasNotificationPermission(this)) {
-            notificationPermissions.requestNotificationPermission(
-                this,
-                ACTIVITY_NOTIFICATION_PERMISSIONS_REQUEST_CODE
-            )
-        } else {
-            setBackgroundJobWithParameters(backgroundParameters)
-        }
+
+        val permissions = mutableListOf<String>().apply {
+            if (SDK_INT >= 33) {
+                add(Manifest.permission.POST_NOTIFICATIONS)
+                add(Manifest.permission.ACTIVITY_RECOGNITION)
+            }
+            else if (SDK_INT >= 29) {
+                add(Manifest.permission.ACTIVITY_RECOGNITION)
```

Comment:
In order to avoid duplication, maybe we could

```
if (SDK_INT >= 33) {
     add(Manifest.permission.POST_NOTIFICATIONS)
}
if (SDK_INT >= 29) {
     add(Manifest.permission.ACTIVITY_RECOGNITION)
}
```

Reply:
My reasoning here was that using the `if-else` would be a bit faster. I know it doesn't make a difference, but still. What do you think? What should we go for? Should we go with the more optimized solution or the one with less code repetition?

Reply:
I believe the maintainability gains are greater.

Reply:
Alright, just wanted to raise the topic.

I agree, and have updated the code accordingly. Can you check?